### PR TITLE
Serve brotli static files

### PR DESCRIPTION
**Pull request by 2-5**
## What do these changes do?

Enable serving of statically compressed Brotli files (.br) if they are present, just like for .gz

## Are there changes in behavior for the user?

If the user has statically compressed .br files present when trying to server a file using FileResponse, they will now be served instead of ignored. The .br files also take priority over .gz files.

## Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by asvetlov** [comment]
Tests for a new functionality are required.

**Comment by 2-5** [comment]
I tried to write tests, but failed to.

For behavioral tests, I looked into `test_web_sendfile.py` but failed to understand how they work. Those tests somehow test if the `.png` or `.png.gz` file is returned, depending on the `Accept-Encoding` headers, yet the `.gz` extension doesn't appear nowhere.

I also tried adding functional tests in `test_web_sendfile_functional.py` similar to the `gzip` ones, but failed again. When I try serving a `.br` file, and request it back with the proper headers (since the default one is `gzip, deflate`):

```python
resp = await client.get('/', headers={'Accept-Encoding':'gzip, deflate, br'})
```
it fails with this error: 
```python
aiohttp.client_exceptions.ClientPayloadError: 400, message='Can not decode content-encoding: br'
```
even if I have the `brotli` package installed.

I also failed writing a test for `app.router.add_static` like the ones at the end of the file.

The manual tests that I did do: use Chrome as a client, `aiohttp.web` with `add_static` as server, and confirm that when a `.br` file is present it's served instead of the `.gz` or the regular one (confirmed by looking at `Content-Type`, `Content-Length` and `Content-Encoding`)

File: CONTRIBUTORS.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 - Contributors -
 ----------------
+2-5
```
Comment (webknjaz) [inline CONTRIBUTORS.txt]: I think we prefer actual name.

File: aiohttp/web_fileresponse.py [hunk 2]
```diff
@@ -222,12 +235,16 @@ def __init__(self, path: Union[str, pathlib.Path],
             return await super().prepare(request)
 
         if hdrs.CONTENT_TYPE not in self.headers:
-            ct, encoding = mimetypes.guess_type(str(filepath))
+            # Guess Content-Type and encoding of original path,
```
Comment (webknjaz) [inline aiohttp/web_fileresponse.py:238]: I think you can register new mime types there.

File: aiohttp/web_fileresponse.py [hunk 3]
```diff
@@ -198,13 +198,26 @@ def __init__(self, path: Union[str, pathlib.Path],
     ) -> Optional[AbstractStreamWriter]:
         filepath = self._path
 
+        encoding = None
+        accept_encoding = request.headers.get(hdrs.ACCEPT_ENCODING, '')
+
+        brotli = False
+        if 'br' in accept_encoding:
```
Comment (webknjaz) [inline aiohttp/web_fileresponse.py:205]: You also need some check for whether brotli dist is installed in the env at all.

File: CONTRIBUTORS.txt (hunk 1 above)
Comment (2-5) [reply CONTRIBUTORS.txt]: Removed, I don't actually want attribution

File: aiohttp/web_fileresponse.py (hunk 3 above)
Comment (2-5) [reply aiohttp/web_fileresponse.py:205]: Not really. We are just serving the raw compressed bytes as they are found on disk. The client requesting them (with indication of support for brotli in the headers) does the decoding. This client might not even be a Python script, like Chrome.

File: aiohttp/web_fileresponse.py (hunk 2 above)
Comment (2-5) [reply aiohttp/web_fileresponse.py:238]: I thought about that, but there is no consensus about the filename/encoding name (see https://bugs.python.org/issue32021). Also, registering mimetypes in request handlers which might be seldom/never called seemed wrong, with a good posibility of creating hard to track down bugs for the user of the library.

**Comment by asvetlov** [comment]
Closing the PR.
1. It fails on CI
2. I never saw `file.br` archive, while `file.gzip` is pretty common. There is no reason for supporting the former.
3. The PR author didn't provide tests for added functionality.

**Comment by 2-5** [comment]
> 2. I never saw `file.br` archive, while `file.gzip` is pretty common. There is no reason for supporting the former.

That is because Brotli files are either statically created by an asset pipeline (webpack/...) or dynamically compressed when serving (nginx/...). You will probably never see a freee one in the wild.

**Comment by asvetlov** [comment]
The lack of unittests is still the main objection.
If you want to finish the PR -- you are welcome

**Pull request closed without merging**